worlds/civ_6/__init__.py

```diff
@@ -0,0 +1,322 @@
+
+from .ProgressiveDistricts import get_flat_progressive_districts
+from worlds.generic.Rules import forbid_item
+
+
+from .Data import (
+    get_boosts_data,
+    get_era_required_items_data,
+)
+
+from .Rules import create_boost_rules
+from .Container import (
+    CivVIContainer,
+    generate_goody_hut_sql,
+    generate_new_items,
+    generate_setup_file,
+    generate_update_boosts_sql,
+)
+from .Enum import CivVICheckType, CivVIHintClassification
+from .Items import (
+    BOOSTSANITY_PROGRESSION_ITEMS,
+    FILLER_DISTRIBUTION,
+    CivVIEvent,
+    CivVIItemData,
+    FillerItemRarity,
+    format_item_name,
+    generate_item_table,
+    CivVIItem,
+    get_item_by_civ_name,
+    get_random_filler_by_rarity,
+)
+from .Locations import (
+    CivVILocation,
+    CivVILocationData,
+    EraType,
+    generate_era_location_table,
+    generate_flat_location_table,
+)
+from .Options import CivVIOptions
+from .Regions import create_regions
+from BaseClasses import Item, ItemClassification, MultiWorld, Tutorial
+from worlds.AutoWorld import World, WebWorld
+from worlds.LauncherComponents import Component, SuffixIdentifier, Type, components, launch_subprocess  # type: ignore
+
+
+def run_client(*args: Any):
+    print("Running Civ6 Client")
+    from .Civ6Client import main  # lazy import
+
+    launch_subprocess(main, name="Civ6Client")
+
+
+components.append(
+    Component(
+        "Civ6 Client",
+        func=run_client,
+        component_type=Type.CLIENT,
+        file_identifier=SuffixIdentifier(".apcivvi"),
+    )
+)
+
+
+class CivVIWeb(WebWorld):
+    tutorials = [
+        Tutorial(
+            "Multiworld Setup Guide",
+            "A guide to setting up Civilization VI for MultiWorld.",
+            "English",
+            "setup_en.md",
+            "setup/en",
+            ["hesto2"],
+        )
+    ]
+    theme = "ocean"
+
+
+class CivVIWorld(World):
+    """
+    Civilization VI is a turn-based strategy video game in which one or more players compete alongside computer-controlled opponents to grow their individual civilization from a small tribe to control the entire planet across several periods of development.
+    """
+
+    game = "Civilization VI"
+    topology_present = False
+    options_dataclass = CivVIOptions
+    options: CivVIOptions  # type: ignore
+
+    web = CivVIWeb()
+
+    item_name_to_id = {item.name: item.code for item in generate_item_table().values()}
+    location_name_to_id = {
+        location.name: location.code
+        for location in generate_flat_location_table().values()
+    }
+
+    item_table: Dict[str, CivVIItemData] = {}
+    location_by_era: Dict[str, Dict[str, CivVILocationData]]
+    required_client_version = (0, 4, 5)
+    location_table: Dict[str, CivVILocationData]
+    era_required_non_progressive_items: Dict[EraType, List[str]] = {}
+    era_required_progressive_items_counts: Dict[EraType, Dict[str, int]] = {}
+    era_required_progressive_era_counts: Dict[EraType, int] = {}

```

Comment:
Two player generations fail a lot right now because of how Python class vars work.

Unlike other languages, if you initialize (`= {}`) a mutable property at the class level, this value will be **shared** between all instances.

You must (re)initialize the variable in `__init__`.

To avoid this problem, it's better to leave members unitialized at the class level, then this can never "accidentally" happen to you".

Reply:
Dang, I had no idea. That's good to know. I need to take a look at my metroidprime apworld for this as well 👀